blog-posts/getting-started-gatsby-next-remix.md

```diff
@@ -0,0 +1,361 @@
+To load the data for a blogpost the function `getStaticProps` is needed. This function now receives `{params}`, which can be used to determine the path and the depending data. Finally, in the actual page component, the data can be accessed via props (here via `{postData}`). The following source code shows how we built our comparison blog project using SSG and the methods just described.
+
+```tsx
+export async function getStaticPaths() {
+    //     paths: [
+    //         {params: {path: 'first-blogpost'}},
+    //         {params: {path: 'second-blogpost'}},
+    //         ...
+    //         {params: {path: 'fifth-blogpost'}}
+    //     ],
+    const paths = getAllPostPaths()
+
+    return {
+        paths,
+        fallback: false
+    }
+
+export async function getStaticProps({params}: { params: PostsData }) {
+    const postData = await getPostData(params.path)
+    return {
+        props: {
+            postData
+        }
+    }
+}
+
+export default function Post({postData}: { postData: PostData }) {
+    return (
+            <div dangerouslySetInnerHTML={{__html: postData.contentHtml}}/>
+    )
+}
+```
+
+
+> 💡 The object returned by `getStaticProps` can include other keys besides `props`, such as `revalidate`. `revalidate` is set to `false` by default, but can also contain the amount in seconds after which a page re-generation can occur. In this way, individual static pages can be changed at a certain interval after being built. This is called [Incremental Static Regeneration (ISR)](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration).
+
+
+#### **Remix**
+
+Remix supports only server side rendering. An example of this can be found in the section below.
+
+## Server Side Rendering (SSR)
+
+With server side rendering (SSR), the HTML is rebuilt for each request. All three frameworks support SSR.
+
+#### **Gatsby**
+
+Besides static site generation (SSG) you can also use server side rendering (SSR) for certain use cases in Gatsby since version 4. For this, the function `getServerData` must be built into a page, in which the data is requested from the server. This data can then be accessed in the actual page component with `serverData`. In the following the example from the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/rendering-options/using-server-side-rendering/) is considered:
+
+```tsx
+import * as React from "react"
+
+const SSRPage = ({ serverData }) => (
+  <main>
+    <h1>SSR Page with Dogs</h1>
+    <img alt="Happy dog" src={serverData.message} />
+  </main>
+)
+
+export default SSRPage
+
+export async function getServerData() {
+  try {
+    const res = await fetch(`https://dog.ceo/api/breeds/image/random`)
+
+    if (!res.ok) {
+      throw new Error(`Response failed`)
+    }
+
+    return {
+      props: await res.json(),
```

Comment:
The `cache-control` header is crucial when doing SSR, therefore I would add it to the example:

```
{
	props: await res.json(),
	headers: {
      // cache the page in browser for 10 seconds, on the CDN for 1 minute and keep it stale for additional 4 mintues
      'Cache-Control': 'public, max-age=10, s-maxage=60, stale-while-revalidate=240',
    }, 
}
```

Reply:
Ok, there is also no `Cache-Control` header in the next example. We should leave it out then and handle this topic in a different blog post.